src/org/scada_lts/dao/migration/mysql/V2_3__FaultsAndAlarms.java

```diff
@@ -30,70 +35,68 @@
+        jdbcTmp.execute("ALTER TABLE dataPoints ADD pointName VARCHAR(250) COMMENT 'copy point name from data';");
+        jdbcTmp.execute("ALTER TABLE dataPoints ADD plcAlarmLevel TINYINT(8) COMMENT '1 - FAULT, 2 - ALARM';");
     }
 
     private void updateDataPointsTable(JdbcTemplate jdbcTmp) throws Exception {
 
-        //Using DataPointVO forces deserialization
-        List<DataPointVO> dataPoints = jdbcTmp.query("SELECT id, data FROM dataPoints", (resultSet, i) -> {
-            try (InputStream inputStream = resultSet.getBinaryStream("data");
-                 ObjectInputStream objectInputStream = new ObjectInputStream(inputStream)) {
-                DataPointVO dataPointVO = (DataPointVO) objectInputStream.readObject();
-                dataPointVO.setId(resultSet.getInt("id"));
-                return dataPointVO;
-            } catch (IOException | ClassNotFoundException ex) {
-                ex.printStackTrace();
-                return null;
+        try {
+            List<DataPointVO> dataPoints = jdbcTmp.query("SELECT id, data FROM dataPoints", (resultSet, i) -> {
+                try (InputStream inputStream = resultSet.getBinaryStream("data");
+                     ObjectInputStream objectInputStream = new ObjectInputStream(inputStream)) {
+                    DataPointVO dataPointVO = (DataPointVO) objectInputStream.readObject();
+                    dataPointVO.setId(resultSet.getInt("id"));
+                    return dataPointVO;
+                } catch (IOException | ClassNotFoundException ex) {
+                    ex.printStackTrace();
+                    return null;
+                }
+            });
+
+            boolean isNull = dataPoints.stream().anyMatch(Objects::isNull);
+            if (isNull) {
+                throw new IllegalStateException("DataPointVO is null!");
             }
-        });
 
-        boolean isNull = dataPoints.stream().anyMatch(Objects::isNull);
-        if(isNull) {
-            throw new IllegalStateException("DataPointVO is null!");
-        }
-
-        for(DataPointVO dataPointPart: dataPoints) {
-            String dataPointName = dataPointPart.getName();
-            int plcAlarmLevel = 0;
-            if(dataPointName.contains(" AL ")) {
-                plcAlarmLevel = 2;
-            }
-            if(dataPointName.contains(" ST ")) {
-                plcAlarmLevel = 1;
+            for (DataPointVO dataPointPart : dataPoints) {
+                String dataPointName = dataPointPart.getName();
+                int plcAlarmLevel = 0;
+                if (dataPointName.contains(" AL ")) {
+                    plcAlarmLevel = 2;
+                }
+                if (dataPointName.contains(" ST ")) {
+                    plcAlarmLevel = 1;
+                }
+                jdbcTmp.update("UPDATE dataPoints SET plcAlarmLevel = ?, pointName = ? WHERE id = ?",
+                        plcAlarmLevel, dataPointName, dataPointPart.getId());
```

Comment:
EmptyResultDataAccessException - this particular exception can only occur when you call this execution method, if you want to do it very correctly, catching this exception should be on this for loop, if you want to break this loop execution.